src/execution.jl

```diff
@@ -334,8 +312,97 @@ end
 
 ## host-side API
 
-const agecache = Dict{UInt, UInt}()
+using Core.Compiler: retrieve_code_info, CodeInfo, MethodInstance, SSAValue, SlotNumber
+using Base: _methods_by_ftype
+
+# HACK: mechanism to generate calls that are not executed, but ensure method invalidation
+const opaque_false = Ref(false)
+function fake_call(f)
+    opaque_false[] || return
+    f(Ref{Any}()[]...)
```

Comment:
This will give you an edge to the methodtable of `f` and only when you are lucky. I suspect that it many cases this will yield a generic call which not insert a dependency edge.

Reply:
Right, but doesn't that also mean that calling a function is inherently a fragile mechanism for adding an edge since it can always be a generic call? Explicitly adding an edge by setting it in `ci` seems like a better approach then, were it not that it does not seem to work (or I am doing something wrong).

Reply:
Yes, when I had to solve this previously, I added a call with the specific arguments. Those did not end up being generic calls since they were always duspatchtuples.